tests/test_dataset.py

```diff
@@ -0,0 +1,741 @@
+        Since this process cannot always be fully reversible, the expected values
+        after the process will be ``values_after_fix`` property.
+
+        Returns
+        -------
+        pd.Series
+            Column values where possible ReverseFeatureOperation introduced
+            values to fix.
+        """
+        return self._values_to_fix
+
+    @property
+    def values_after_fix(self) -> pd.Series:
+        """
+        Return the column values resulting after applying proper correction.
+
+        These are modified along with ``values_to_fix`` to keep
+        track of the values inserted.
+        Since the process of properly correcting ``values_to_fix`` cannot always
+        be fully reversible, these are the column values that the column
+        will contain after that process.
+
+        Returns
+        -------
+        pd.Series
+            Column values resulting after applying proper correction to
+            ``values_to_fix`` property of the instance.
+        """
+        return self._values_after_fix
+
+    def update_sample(self, index: int, value_to_fix: Any, value_after_fix: Any):
+        """
+        Update a single column sample with new values
+
+        Parameters
+        ----------
+        index: int
+            Index of the column sample that is updated
+        value_to_fix: Any
+            Raw value that needs a fix (usually an error simulated by a
+            ReverseFeatureOperation).
+        value_to_fix: Any
+            Value that is supposed to be found after that the proper correction
+            is applied to ``value_to_fix`` argument.
+        """
+        self._values_to_fix[index] = value_to_fix
+        self._values_after_fix[index] = value_after_fix
+
+    def update_column_values(
+        self, values_to_fix: Iterable[Any], values_after_fix: Iterable[Any]
+    ):
+        """
+        Update a single column sample with new values
+
+        Parameters
+        ----------
+        values_to_fix: Iterable[Any]
+            Raw values that need a fix (usually values containing errors that have
+            been simulated by a ReverseFeatureOperation).
+        values_to_fix: Iterable[Any]
+            Values that are supposed to be found after that the proper correction
+            is applied to ``values_to_fix`` argument.
+        """
+        self._values_to_fix = values_to_fix
+        self._values_after_fix = values_after_fix
+
+
+class TestDataSet:
+    def __init__(self, sample_size: int = None):
+        # Creating two dicts with the same elements (passed by address), so that
+        # retrieving an element by name or index takes the same time
+        self._columns_by_index = []
+        self._sample_size = sample_size
+        self._name_to_index_map = {}
+
+    @property
+    def sample_size(self) -> Union[int, None]:
+        return self._sample_size
+
+    def _validate_testcolumn(
+        self, column: Union[TestColumn, _TestColumn], is_updating: bool = False
```

Comment:
you could change `is_updating` with `existing_column`

Reply:
I split the two methods since they were different except when checking the sample size